### ISSUE TITLE
False positive: Property 'open' does not exist on type 'HTMLElement' for <details> element

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#87]
When using an attachment/action callback on a `<details>` element, svelte-check-rs types the node as `HTMLElement` instead of `HTMLDetailsElement`, causing a false positive when accessing the `open` property.

## Reproduction

```svelte
<script lang="ts">
  function click_outside(opts: { callback: (node: HTMLElement) => void }) {
    return (node: HTMLElement) => {
      const handle_click = (event: MouseEvent) => {
        if (!node.contains(event.target as Node)) {
          opts.callback(node)
        }
      }
      document.addEventListener('click', handle_click)
      return {
        destroy: () => document.removeEventListener('click', handle_click)
      }
    }
  }
</script>

<details {@attach click_outside({ callback: (node) => node.open = false })}>
  <summary>Click me</summary>
  <p>Content</p>
</details>
```

## Error

```
Error: Property 'open' does not exist on type 'HTMLElement'. (ts(TS2339))
```

## Workaround

Cast the node: `(node as HTMLDetailsElement).open = false`

svelte-check-rs: 0.7.1
